core/src/main/java/org/apache/gravitino/listener/ModelEventDispatcher.java

```diff
@@ -0,0 +1,386 @@
+   * model is registered, users can call {@link ModelCatalog#linkModelVersion(NameIdentifier,
+   * String, String[], String, Map)} to link the model version to the registered {@link Model}.
+   *
+   * @param ident The name identifier of the model.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param properties The properties of the model. The properties are optional and can be null or
+   *     empty.
+   * @return The registered model object.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws ModelAlreadyExistsException If the model already registered.
+   */
+  @Override
+  public Model registerModel(NameIdentifier ident, String comment, Map<String, String> properties)
+      throws NoSuchSchemaException, ModelAlreadyExistsException {
+    ModelInfo registerRequest = new ModelInfo(ident.name(), properties, comment);
+    eventBus.dispatchEvent(
+        new RegisterModelPreEvent(PrincipalUtils.getCurrentUserName(), ident, registerRequest));
+    try {
+      Model model = dispatcher.registerModel(ident, comment, properties);
+      // TODO: ModelEvent
+      return model;
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /**
+   * Register a model in the catalog if the model is not existed, otherwise the {@link
+   * ModelAlreadyExistsException} will be thrown. The {@link Model} object will be created when the
+   * model is registered, in the meantime, the model version (version 0) will also be created and
+   * linked to the registered model.
+   *
+   * @param ident The name identifier of the model.
+   * @param uri The model artifact URI.
+   * @param aliases The aliases of the model version. The aliases should be unique in this model,
+   *     otherwise the {@link ModelVersionAliasesAlreadyExistException} will be thrown. The aliases
+   *     are optional and can be empty. Also, be aware that the alias cannot be a number or a number
+   *     string.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param properties The properties of the model. The properties are optional and can be null or
+   *     empty.
+   * @return The registered model object.
+   * @throws NoSuchSchemaException If the schema does not exist when register a model.
+   * @throws ModelAlreadyExistsException If the model already registered.
+   * @throws ModelVersionAliasesAlreadyExistException If the aliases already exist in the model.
+   */
+  @Override
+  public Model registerModel(
+      NameIdentifier ident,
+      String uri,
+      String[] aliases,
+      String comment,
+      Map<String, String> properties)
+      throws NoSuchSchemaException, ModelAlreadyExistsException,
+          ModelVersionAliasesAlreadyExistException {
+    ModelInfo modelInfo = new ModelInfo(ident.name(), properties, comment);
```

Comment:
How about using a new `RegisterAndLinkModelEvent` to represent the operation?  you could pass `ModelInfo` and `aliases` to the event